src/OnboardingSPA/steps/SiteGen/Editor/Header/center.js

```diff
@@ -140,14 +123,89 @@ const StepEditorHeaderCenter = ( {
 						/>
 					</div>
 				) }
-				renderContent={ DropDownMenu }
+				renderContent={ () => (
+					<DropDownMenu
+						onRegenerate={ onRegenerate }
+						onCustomize={ onCustomize }
+						onRenameItemSelect={ onRenameItemSelect }
+						onViewAll={ onViewAll }
+						isLargeViewport={ isLargeViewport }
+					/>
+				) }
 			/>
 		);
+	}
+);
+
+const StepEditorHeaderCenter = ( {
+	handleFavorite,
+	handleRename,
+	handleViewAll,
+	handleRegenerate,
+	handleCustomize,
+	homepageTitle,
+	isFavorite,
+} ) => {
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const inputRef = useRef( null );
+	const [ isInputEnabled, setInputEnabled ] = useState( false );
+	const enableInput = ( isEnabled ) => {
+		setInputEnabled( isEnabled );
+		if ( isEnabled ) {
+			inputRef.current?.focus();
+		}
+	};
+
+	const onRegenerate = () => {
+		if ( typeof handleRegenerate === 'function' ) {
+			return handleRegenerate();
+		}
+	};
+
+	const onFavorite = () => {
+		if ( typeof onFavorite === 'function' ) {
+			return handleFavorite();
+		}
+	};
+
+	const onCustomize = () => {
+		if ( typeof handleCustomize === 'function' ) {
+			return handleCustomize();
+		}
+	};
+
+	const onViewAll = () => {
+		if ( typeof handleViewAll === 'function' ) {
+			return handleViewAll();
+		}
+	};
+
+	const onRenameItemSelect = () => {
+		enableInput( true );
+	};
+
+	const onRename = () => {
+		if ( typeof handleRename === 'function' ) {
+			handleRename( inputRef.current.value );
```

Comment:
Should we change the data in homepages if rename was selected and never edited? That does cause an unnecessary re-render. Also should we trim the data?